src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -53,81 +53,57 @@ internal EventGridPublisher(string topicEndpoint, string authenticationKey, Poli
         public string TopicEndpoint { get; }
 
         /// <summary>
-        ///     Creates a validated EventGrid publisher
+        ///     Publish an event grid message to the configured Event Grid topic
         /// </summary>
-        /// <param name="topicEndpoint">Url of the custom Event Grid topic</param>
-        /// <param name="authenticationKey">Authentication key for the custom Event Grid topic</param>
-        [Obsolete("Use the EventGridPublisherBuilder instead to create EventGridPublisher models")]
-        public static EventGridPublisher Create(string topicEndpoint, string authenticationKey)
+        /// <typeparam name="TEvent">Type of the specific EventData</typeparam>
+        /// <param name="event">Event to publish</param>
+        public async Task Publish<TEvent>(TEvent @event)
+            where TEvent : class, IEvent, new()
         {
-            Guard.NotNullOrWhitespace(topicEndpoint, nameof(topicEndpoint), "The topic endpoint must not be empty and is required");
-            Guard.NotNullOrWhitespace(authenticationKey, nameof(authenticationKey), "The authentication key must not be empty and is required");
+            Guard.NotNull(@event, nameof(@event), "No event was specified");
+
+            List<TEvent> eventList = new List<TEvent>
+            {
+                @event
+            };
 
-            var eventGridPublisher = new EventGridPublisher(topicEndpoint, authenticationKey);
-            return eventGridPublisher;
+            await PublishEventToTopic(eventList);
         }
 
         /// <summary>
         ///     Publish an event grid message to the configured Event Grid topic
         /// </summary>
-        /// <typeparam name="TData">Type of the specific EventData</typeparam>
-        /// <param name="subject">Subject</param>
-        /// <param name="eventType">Event Type</param>
-        /// <param name="data">Typed event data</param>
-        /// <param name="id">Unique id of the event, generated by default</param>
-        /// <returns></returns>
-        public async Task Publish<TData>(string subject, string eventType, IEnumerable<TData> data, string id = null)
+        /// <typeparam name="TEvent">Type of the specific EventData</typeparam>
+        /// <param name="events">Events to publish</param>
+        public async Task Publish<TEvent>(List<TEvent> events)
```

Comment:
Is there a reason this is a `List<TEvent>`? (similar for other overloads)

```suggestion
        public async Task Publish<TEvent>(IEnumerable<TEvent> events)
```

Reply:
Not really, just a (bad) habit

Reply:
So this is kinda funny. If I switch this to IEnumerable it thinks that List<TEvent> is a single event and it cannot cast it to TEvent.

I would suggest to either keep it as is or rename `Publish<TEvent>(IEnumerable<TEvent> events)` to `PublishMany<TEvent>(IEnumerable<TEvent> events)`

Reply:
Thoughts @samneirinck/@stijnmoreels?

Reply:
Prefer option 2, since if I pass in an `IEnumerable<T>`/`T[]`/,,,, I would not get the behavior I expected.

Reply:
Agreed, changed this